lib-driver/caqti_driver_mariadb.ml

```diff
@@ -493,6 +495,39 @@ module Connect_functor (System : Caqti_driver_sig.System_unix) = struct
       (function
        | Ok () -> return (Ok ())
        | Error err -> transaction_failed "# ROLLBACK" err)
+
+    let populate ~table ~columns row_type input_stream =
+      let columns_tuple = "(" ^ (String.concat "," columns) ^ ")" in
+      let values_tuple = "(" ^ (String.concat "," (List.map (fun _ -> "?") columns)) ^ ")" in
+      let insert_query =
+        Caqti_request.exec
+          row_type
+          ("INSERT INTO " ^ table ^  " " ^ columns_tuple ^ " VALUES " ^ values_tuple)
+      in
```

Comment:
You need to pass `~oneshot:true` here to avoid accumulating prepared queries for each call.